### PR TITLE
Improve shadow domain support

### PR DESCRIPTION
**Pull request by JoosuaKoskinen**
#### What are the main changes in this PR?

- Adds Seravo Plugin side support for shadows with domain (more in commit message)
- Makes the functionality of instance switcher more constant

##### Why are we doing this? Any context or related work?

Problem reported in issue #67

#### Where should a reviewer start?

Reviewer should try accessing and exiting different shadows in different ways (url / instance-switcher / shadow banner). Site 'joosuakoskinen' in Sestak can be used for testing. It has 2 shadows, one with domain and one without.

### CODE REVIEW

File: js/instance-switcher.js [hunk 1]
```diff
@@ -1,30 +1,82 @@
...
-  var limit = links.length;
+    // Reload / redirect page
+    location.href = new_location;
+  }
+
+  // Add right listener for the links found
+
+  var limit = shadow_links.length;
+  for (var i = 0; i < limit; i++) {
+      shadow_links[i].addEventListener('click', shadow_listener);
```
Comment (ottok) [inline js/instance-switcher.js]: Indentation 2 -> 4 ?

File: js/instance-switcher.js [hunk 2]
```diff
@@ -1,30 +1,82 @@
...
+  }
+
+  limit = exit_links.length;
   for (var i = 0; i < limit; i++) {
-    links[i].addEventListener('click', listener);
+      exit_links[i].addEventListener('click', exit_listener);
```
Comment (ottok) [inline js/instance-switcher.js]: Indentation 2 -> 4 ?

File: js/instance-switcher.js (hunk 1 above)
Comment (JoosuaKoskinen) [reply js/instance-switcher.js]: Fixed back to 2

File: js/instance-switcher.js (hunk 2 above)
Comment (JoosuaKoskinen) [reply js/instance-switcher.js]: Also fixed.

**Comment by JoosuaKoskinen** [comment]
> The commit message is really good! Since the text does not describe only the change, but mostly how the code now permanently works, should some parts of the commit message maybe be as inline comments in the code?

The same fallback flow is commented in `instance-switcher.php`.

File: modules/instance-switcher.php [hunk 3]
```diff
@@ -193,6 +216,43 @@ public static function render_shadow_admin_notice( $current_screen ) {
         echo $admin_notice_content;
       }
     }
+
+    public static function get_shadow_domain( $identifier ) {
+      $domain_list = API::get_site_data('/domains');
+      if ( is_wp_error( $domain_list ) ) {
+        die( $domain_list->get_error_message() );
```
Comment (elguitar) [inline modules/instance-switcher.php]: The error from the API should be handled gracefully. Dying at this point leads to wp-admin not drawing pages.

Comment (JoosuaKoskinen) [reply modules/instance-switcher.php]: Fixed, instead of `die()` this now returns null and the caller handles that properly.

**Comment by elguitar** [comment]
Apart from failing CI tests (identation and whitespace), looks good to me!

**Pull request merged**